src/mapbox/Map.js

```diff
@@ -14,10 +15,16 @@ class Map extends mixin(MBMap) {
    * Constructor.
    */
   constructor(options) {
+    const { style } = options;
+    const apiKey = options.apiKey || false;
+    const apiKeyName = options.apiKeyName || 'key';
+    let newStyle = options.style;
+    newStyle = getMapboxStyle(apiKey, apiKeyName, style);
```

Comment:
In getMapboxStyle you should manage the case where options.style is already a mapbox style object  and not a url

Reply:
The way it is used in the constructor don't prevent this to happen actually?